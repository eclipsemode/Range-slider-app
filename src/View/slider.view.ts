import {ClassName, ModelOption, ControlsEnum, TogglesEnum} from '../utils';
import Observer from '../Observer/Observer';

import {
    setBar,
    setSlider,
    setRulers,
    setRange,
    setTooltip,
    setColor,
    setVertical,
    setControl,
    setConfig
} from './SubViews';
import {
    observeThumbs,
    observeControl,
    observeConfig
} from '../Observer';
import $ from 'jquery';


class View extends Observer {
    private readonly selectorState: string;
    private readonly optionsState: ModelOption;
    private readonly setSlider: CallableFunction;
    private readonly setRange: CallableFunction;
    private readonly setRulers: CallableFunction;
    private readonly setBar: CallableFunction;
    private readonly setTooltip: CallableFunction;
    private readonly setColor: CallableFunction;
    private readonly setVertical: CallableFunction;
    private readonly setControl: CallableFunction;
    private readonly setConfig: CallableFunction;
    private readonly observeThumbs: CallableFunction;
    private readonly observeControl: CallableFunction;
    private readonly observeConfig: CallableFunction;

    constructor(private selector: string, private options: ModelOption) {
        super();
        this.selectorState = selector;
        this.optionsState = options;
        this.setSlider = setSlider.bind(this);
        this.setRulers = setRulers.bind(this);
        this.setRange = setRange.bind(this);
        this.setBar = setBar.bind(this);
        this.setTooltip = setTooltip.bind(this);
        this.setColor = setColor.bind(this);
        this.setVertical = setVertical.bind(this);
        this.setControl = setControl.bind(this);
        this.setConfig = setConfig.bind(this);
        this.observeThumbs = observeThumbs.bind(this);
        this.observeControl = observeControl.bind(this);
        this.observeConfig = observeConfig.bind(this);
    }

    render(): void {
        this.opts = this.optionsState;
        this.setSlider();
        this.setBar();
        this.setRange();
        this.setRulers();
        this.setColor();
        this.setControl();
        this.setConfig();
        this.setTooltip();
        this.setVertical();
        this.thumbsObserver();
        this.controlObserver();
        this.configObserver();
    }

    thumbsObserver() {
        const ObserveThumbs = this.observeThumbs();
        const $thumbs: JQuery = $('.js-slider-app__input');
        const $thumbMin = $(this.selectorState + ' ' + ClassName.MIN);
        const $thumbMax = $(this.selectorState + ' ' + ClassName.MAX);

        $thumbs.off();
        $thumbs.on('input', (e: JQuery.ChangeEvent) => {
            if ($(e.currentTarget).hasClass('js-slider-app__input-min')) {
                if (+$thumbMin.val() > +$thumbMax.val() - this.opts.gap && this.opts.range) {
                    +$thumbMax.val() - this.opts.step < +$thumbMax.val() - this.opts.gap
                        ? $thumbMin.val(+$thumbMax.val() - this.opts.step)
                        : $thumbMin.val(+$thumbMax.val() - this.opts.gap);
                }

                ObserveThumbs.opts = {
                    ...this.opts,
                    from: +e.target.value
                };
            } else {
                if (+$thumbMin.val() > +$thumbMax.val() - this.opts.gap) {
                    +$thumbMin.val() + this.opts.step > +$thumbMin.val() + this.opts.gap
                        ? $thumbMax.val(+$thumbMin.val() + this.opts.step)
                        : $thumbMax.val(+$thumbMin.val() + this.opts.gap);
                }

                ObserveThumbs.opts = {
                    ...this.opts,
                    to: +e.target.value
                };
            }
        });
    }

    controlObserver() {
        const ObserveControl = this.observeControl();
        this.opts.controlConfig.forEach((item: string): void => {
            const $element: JQuery = $(`${ this.selectorState }__control-${ item }`);

            $element.off();
            $element.on('change', (e: JQuery.ChangeEvent) => {
                const $thumbMin = $(this.selectorState + ' ' + ClassName.MIN);
                const $thumbMax = $(this.selectorState + ' ' + ClassName.MAX);
                switch (item) {
                    case ControlsEnum.MIN:
                        ObserveControl.opts = {
                            ...this.opts,
                            min: +e.target.value
                        };
                        break;
                    case ControlsEnum.MAX:
                        ObserveControl.opts = {
                            ...this.opts,
                            max: +e.target.value
                        };
                        break;
                    case ControlsEnum.STEP:
                        ObserveControl.opts = {
                            ...this.opts,
                            step: +e.target.value
                        };
                        break;
                    case ControlsEnum.FROM:
                        if (+e.target.value > +$thumbMax.val() - this.opts.gap && this.opts.range) {
                            +$thumbMax.val() - this.opts.step < +$thumbMax.val() - this.opts.gap
                                ? e.target.value = +$thumbMax.val() - this.opts.step
                                : e.target.value = +$thumbMax.val() - this.opts.gap;
                        }

                        ObserveControl.opts = {
                            ...this.opts,
                            from: +e.target.value
                        };
                        break;
                    case ControlsEnum.TO:
                        if (+$thumbMin.val() > +e.target.value - this.opts.gap) {
                            +$thumbMin.val() + this.opts.step > +$thumbMin.val() + this.opts.gap
                                ? e.target.value = +$thumbMin.val() + this.opts.step
                                : e.target.value = +$thumbMin.val() + this.opts.gap;
                        }

                        ObserveControl.opts = {
                            ...this.opts,
                            to: +e.target.value
                        };
                        break;
                }
            });
        });
    }

    configObserver() {
        const ObserveConfig = this.observeConfig();

        this.opts.toggleConfig.forEach((item: string): void => {
            const $element: JQuery = $(`${ this.selectorState }__toggle-${ item }`);

            $element.off();
            $element.on('change', (e: JQuery.ChangeEvent) => {
                switch (item) {
                    case TogglesEnum.VERTICAL:
                        ObserveConfig.opts = {
                            ...this.opts,
                            vertical: e.target.checked
                        };
                        break;
                    case TogglesEnum.RANGE:
                        ObserveConfig.opts = {
                            ...this.opts,
                            range: e.target.checked
                        };
                        break;
                    case TogglesEnum.RULERS:
                        ObserveConfig.opts = {
                            ...this.opts,
                            rulers: e.target.checked
                        };
                        break;
                    case TogglesEnum.PROGRESS:
                        ObserveConfig.opts = {
                            ...this.opts,
                            progress: e.target.checked
                        };
                        break;
                    case TogglesEnum.TOOLTIP:
                        ObserveConfig.opts = {
                            ...this.opts,
                            tooltip: e.target.checked
                        };
                        break;
                }
            });
        });
    }
}

export default View;