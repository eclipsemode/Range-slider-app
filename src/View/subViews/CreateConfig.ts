class CreateConfig {
  public configElement: JQuery;

  private toggleElement: JQuery;

  private controlElement: JQuery;

  constructor(private parent: JQuery, private readonly toggleConfig: string[]) {
    this.createToggles(parent, toggleConfig);
  }

  private createToggles(parent: JQuery, toggleConfig: string[]) {
    this.configElement = $("<div>", {
      class: "slider-app__config",
    });
    this.toggleElement = $("<div>", {
      class: "slider-app__config-toggle",
    });
    this.configElement.append(this.toggleElement);

    toggleConfig.forEach((name) => {
      const selectorByName = `slider-app__config--${name}`;

      this.toggleElement.append(
        $(`
                <div class="slider-app__config-element">
                    <label class="slider-app__config-name" for=${selectorByName}>${name}</label>
                    <label class="slider-app__config-container">
                        <input class='slider-app__config-toggle-box ${selectorByName}' type="checkbox">
                        <span class="slider-app__config-toggle-btn"></span>
                    </label>
                </div>
      `)
      );
    });
    this.configElement.appendTo(parent);
  }
}

export default CreateConfig;
