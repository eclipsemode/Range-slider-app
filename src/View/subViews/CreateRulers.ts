class CreateRulers {
  public rulersElement: JQuery;

  constructor(parent: JQuery, valuesArr: number[], pixelsArr: number[]) {
    this.createRulers(parent, valuesArr, pixelsArr);
  }

  private createRulers(
    parent: JQuery,
    valuesArr: number[],
    pixelArr: number[]
  ) {
    this.rulersElement = $("<div>", {
      class: "slider-app__rulers",
    });

    for (let i = 0; i < valuesArr.length; i++) {
      this.rulersElement.append(
        $("<div>", {
          class: `slider-app__rulers-value slider-app__rulers-value--${pixelArr[i]}`,
          text: valuesArr[i],
          css: {
            left: `${pixelArr[i]}px`,
          },
        })
      );
    }

    this.rulersElement.appendTo(parent);
  }
}

export default CreateRulers;
