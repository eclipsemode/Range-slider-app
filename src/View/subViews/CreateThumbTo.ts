import $ from "jquery";

class CreateThumbTo {
  public toThumbElement: JQuery;

  constructor(private readonly parent: JQuery) {
    this.createToThumb(parent);
  }

  private createToThumb(parent: JQuery): void {
    this.toThumbElement = $("<div>", {
      class: "slider-app__thumb slider-app__thumb-to",
    });
    this.toThumbElement.appendTo(parent);
  }
}

export default CreateThumbTo;
