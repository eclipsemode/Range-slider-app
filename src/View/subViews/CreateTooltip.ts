class CreateTooltip {
  public tooltipElement: JQuery;

  constructor(private readonly parent: JQuery) {
    this.createTooltip(parent);
  }

  private createTooltip(parent: JQuery): void {
    this.tooltipElement = $("<div>", {
      class: "slider-app__tooltip",
      text: "100",
    });

    this.tooltipElement.prependTo(parent);
  }
}

export default CreateTooltip;
