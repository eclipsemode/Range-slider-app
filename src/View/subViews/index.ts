export { default as CreateBar } from "./bar/CreateBar";

export { default as CreateConfig } from "./config/CreateConfig";
export { default as RenderConfig } from "./config/RenderConfig";

export { default as CreateProgress } from "./progress/CreateProgress";
export { default as RenderProgress } from "./progress/RenderProgress";

export { default as CreateRulers } from "./rulers/CreateRulers";
export { default as RenderRulers } from "./rulers/RenderRulers";

export { default as CreateThumbFrom } from "./thumbs/CreateThumbFrom";
export { default as CreateThumbTo } from "./thumbs/CreateThumbTo";
export { default as RenderThumbs } from "./thumbs/RenderThumbs";

export { default as CreateTooltip } from "./tooltip/CreateTooltip";
export { default as RenderTooltip } from "./tooltip/RenderTooltip";
