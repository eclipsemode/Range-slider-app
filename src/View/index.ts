import $ from "jquery";
import {
  ModelOption,
  convertToNumber,
  ControlsEnum,
  ActionEnum,
} from "../utils";

import CreateBar from "./subViews/CreateBar";
import CreateThumbFrom from "./subViews/CreateThumbFrom";
import CreateThumbTo from "./subViews/CreateThumbTo";
import CreateProgress from "./subViews/CreateProgress";
import CreateTooltip from "./subViews/CreateTooltip";
import CreateRulers from "./subViews/CreateRulers";
import CreateConfig from "./subViews/CreateConfig";

function calcMouseOffset(mouseOffset: number, sliderWidth: number): number {
  if (mouseOffset < 0) {
    return 0;
  }
  if (mouseOffset > sliderWidth) {
    return sliderWidth;
  }
  return mouseOffset;
}

function convertToPixel(
  value: number,
  sliderWidth: number,
  min: number,
  max: number
): number {
  if (min < 0) {
    const newMax: number = max + Math.abs(min);
    const newValue: number = value + Math.abs(min);
    return Math.round((newValue / newMax) * sliderWidth);
  }
  if (min > 0) {
    const newMax: number = max - min;
    const newValue: number = value - min;
    return Math.round((newValue / newMax) * sliderWidth);
  }
  return Math.round((value / max) * sliderWidth);
}

class View {
  private readonly app: JQuery;

  private readonly bar: CreateBar;

  private fromThumb: CreateThumbFrom;

  private toThumb: CreateThumbTo;

  private progress: CreateProgress;

  private tooltipFrom: CreateTooltip;

  private tooltipTo: CreateTooltip;

  private rulers: CreateRulers;

  private config: CreateConfig;

  private options: ModelOption;

  constructor(private readonly selector: string) {
    this.app = $(selector);

    this.bar = new CreateBar(this.app);
    this.fromThumb = new CreateThumbFrom(this.bar.barElement);
  }

  public bindChangeOptions(handler: CallableFunction) {
    /**
     * Binds thumbs.
     */

    this.app.on("mousedown", (e: JQuery.MouseDownEvent) => {
      const sliderWidth: number = this.bar.barElement.innerWidth();
      const sliderLeftOffset: number = this.bar.barElement.offset().left;

      if (
        e.target.classList.contains(
          this.fromThumb.fromThumbElement[0].classList[1]
        )
      ) {
        $(e.target).on("dragstart", () => false);

        // eslint-disable-next-line @typescript-eslint/no-shadow
        const moveAt = (e: JQuery.MouseMoveEvent | JQuery.MouseDownEvent) => {
          const mouseOffset: number = e.pageX - sliderLeftOffset;
          const thumbOffsetValue: number = calcMouseOffset(
            mouseOffset,
            sliderWidth
          );

          this.options.from = convertToNumber(
            thumbOffsetValue,
            sliderWidth,
            this.options.min,
            this.options.max
          );

          handler(this.options, ActionEnum.DRAG_FROM);
        };

        moveAt(e);

        // eslint-disable-next-line @typescript-eslint/no-shadow
        $(document).on("mousemove", (e) => moveAt(e));

        $(e.target).on("mouseleave", () => {
          $(document).on("mouseup", () => {
            $(document).off("mousemove");
            $(document).off("mouseup");
            $(e.target).off("mouseleave");
          });
        });
        $(e.target).on("mouseup", () => {
          $(document).off("mousemove");
          $(e.target).off("mouseup");
        });
      }

      if (
        e.target.classList.contains(
          this.options.range && this.toThumb.toThumbElement[0].classList[1]
        )
      ) {
        $(e.target).on("dragstart", () => false);

        // eslint-disable-next-line @typescript-eslint/no-shadow
        const moveAt = (e: JQuery.MouseMoveEvent | JQuery.MouseDownEvent) => {
          const mouseOffset: number = e.pageX - sliderLeftOffset;
          const thumbOffsetValue: number = calcMouseOffset(
            mouseOffset,
            sliderWidth
          );

          this.options.to = convertToNumber(
            thumbOffsetValue,
            sliderWidth,
            this.options.min,
            this.options.max
          );

          handler(this.options, ActionEnum.DRAG_TO);
        };

        moveAt(e);

        $(document).on(
          "mousemove",
          // eslint-disable-next-line @typescript-eslint/no-shadow
          (e: JQuery.MouseMoveEvent | JQuery.MouseDownEvent) => moveAt(e)
        );

        $(e.target).on("mouseleave", () => {
          $(document).on("mouseup", () => {
            $(document).off("mousemove");
            $(document).off("mouseup");
            $(e.target).off("mouseleave");
          });
        });
        $(e.target).on("mouseup", () => {
          $(document).off("mousemove");
          $(e.target).off("mouseup");
        });
      }
    });

    /**
     * Binds thumbs move.
     */

    this.bar?.barElement.on("click", (e) => {
      const sliderWidth: number = this.bar.barElement.innerWidth();
      const sliderLeftOffset: number = this.bar.barElement.offset().left;
      const clickedOffset: number = e.pageX - sliderLeftOffset;
      const valueCalculated: number = calcMouseOffset(
        clickedOffset,
        sliderWidth
      );
      const convertedValue: number = convertToNumber(
        valueCalculated,
        sliderWidth,
        this.options.min,
        this.options.max
      );

      if (this.options.range) {
        if (
          Math.abs(convertedValue - this.options.from) <
          Math.abs(convertedValue - this.options.to)
        ) {
          this.options.from = convertedValue;
          handler(this.options, ActionEnum.CLICK_FROM);
        } else {
          this.options.to = convertedValue;
          handler(this.options, ActionEnum.CLICK_TO);
        }
      } else {
        this.options.from = convertedValue;
        handler(this.options);
      }
    });

    /**
     * Binds rulers values.
     */

    $(this.app).on("click", (e) => {
      if (e.target.classList.contains("slider-app__rulers-value")) {
        if (this.options.range) {
          if (
            Math.abs(+e.target.textContent - this.options.from) <
            Math.abs(+e.target.textContent - this.options.to)
          ) {
            this.options.from = +e.target.textContent;
          } else {
            this.options.to = +e.target.textContent;
          }
        } else {
          this.options.from = +e.target.textContent;
        }
      }
      handler(this.options);
    });

    /**
     * Binds toggles.
     */

    this.app.on("input", (e) => {
      this.options.toggleConfig.forEach((name) => {
        if (e.target.classList.contains(`slider-app__toggle--${name}`)) {
          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
          // @ts-ignore
          this.options[name] = $(`.slider-app__toggle--${name}`).prop(
            "checked"
          );
        }
      });

      this.options.controlConfig.forEach((name) => {
        const element: JQuery = $(`.slider-app__control--${name}`);
        if (e.target.classList.contains(`slider-app__control--${name}`)) {
          if (name === ControlsEnum.FROM) {
            if (element.val() > this.options.to - this.options.gap) {
              element.val(this.options.to - this.options.gap);
            }
          }
          if (name === ControlsEnum.TO) {
            if (element.val() < this.options.from + this.options.gap) {
              element.val(this.options.from + this.options.gap);
            }
          }
          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
          // @ts-ignore
          this.options[name] = element.val();
        }
      });
      handler(this.options);
    });

    // this.app.on("input", (e) => {
    //   if (e.target.classList.contains("slider-app__control--from")) {
    //     if (
    //       +$(`.slider-app__control--from`).val() >
    //       this.options.to - this.options.gap
    //     ) {
    //       $(`.slider-app__control--from`).val(
    //         this.options.to - this.options.gap
    //       );
    //     }
    //     this.options.from = +$(`.slider-app__control--from`).val();
    //   }
    // });

    /**
     * Binds resize window.
     */

    $(window).on("resize", () => handler(this.options));
  }

  public render(options: ModelOption) {
    this.options = options;

    /**
     * Creates Tooltip.
     */
    if (this.options.tooltip) {
      const sliderWidth: number = this.bar.barElement.innerWidth();
      if (!this.tooltipFrom) {
        this.tooltipFrom = new CreateTooltip(this.bar.barElement);
      }
      this.bar.barElement.css("margin-top", "2em");
      this.tooltipFrom.tooltipElement.css(
        "left",
        `${convertToPixel(
          this.options.from,
          sliderWidth,
          this.options.min,
          this.options.max
        )}px`
      );
      this.tooltipFrom.tooltipElement.text(this.options.from);

      if (this.options.range) {
        if (!this.tooltipTo) {
          this.tooltipTo = new CreateTooltip(this.bar.barElement);
        }

        this.tooltipTo.tooltipElement.css(
          "left",
          `${convertToPixel(
            this.options.to,
            sliderWidth,
            this.options.min,
            this.options.max
          )}px`
        );
        this.tooltipTo.tooltipElement.text(this.options.to);
      } else if (this.tooltipTo) {
        this.tooltipTo.tooltipElement.remove();
        this.tooltipTo = null;
      }
    } else {
      if (this.tooltipFrom) {
        this.tooltipFrom.tooltipElement.remove();
        this.tooltipFrom = null;
      }

      if (this.tooltipTo) {
        this.tooltipTo.tooltipElement.remove();
        this.tooltipTo = null;
      }

      this.bar.barElement.css("margin-top", "auto");
      // this.app.css("padding", "1em 1.5em 1em");
    }

    /**
     * Creates Progress.
     */
    if (this.options.progress) {
      if (!this.progress) {
        this.progress = new CreateProgress(this.bar.barElement);
      }

      if (!this.options.range) {
        this.progress.progressElement.css("left", "auto");
      }
    } else if (this.progress) {
      this.progress.progressElement.remove();
      this.progress = null;
    }

    /**
     * Changes css of thumb position.
     */
    if (this.options.range) {
      const sliderWidth: number = this.bar.barElement.innerWidth();
      if (!this.toThumb) {
        this.toThumb = new CreateThumbTo(this.bar.barElement);
      }

      this.fromThumb.fromThumbElement.css(
        "left",
        `${convertToPixel(
          this.options.from,
          sliderWidth,
          this.options.min,
          this.options.max
        )}px`
      );

      this.toThumb.toThumbElement.css(
        "left",
        `${convertToPixel(
          this.options.to,
          sliderWidth,
          this.options.min,
          this.options.max
        )}px`
      );

      if (this.progress?.progressElement) {
        this.progress.progressElement.css(
          "left",
          `${convertToPixel(
            this.options.from,
            sliderWidth,
            this.options.min,
            this.options.max
          )}px`
        );

        this.progress.progressElement.css(
          "width",
          `${
            convertToPixel(
              this.options.to,
              sliderWidth,
              this.options.min,
              this.options.max
            ) -
            convertToPixel(
              this.options.from,
              sliderWidth,
              this.options.min,
              this.options.max
            )
          }px`
        );
      }
    } else {
      const sliderWidth: number = this.bar.barElement.innerWidth();

      if (this.toThumb) {
        this.toThumb.toThumbElement.remove();
        this.toThumb = null;
      }

      this.fromThumb.fromThumbElement.css(
        "left",
        `${convertToPixel(
          this.options.from,
          sliderWidth,
          this.options.min,
          this.options.max
        )}px`
      );

      if (this.progress?.progressElement) {
        this.progress.progressElement.css(
          "width",
          `${convertToPixel(
            this.options.from,
            sliderWidth,
            this.options.min,
            this.options.max
          )}px`
        );
      }
    }

    /**
     * Creates Rulers.
     */

    if (this.options.rulers) {
      if (this.rulers) {
        this.rulers.rulersElement?.remove();
        this.rulers = null;
      }

      if (!this.rulers) {
        const sliderWidth: number = this.bar.barElement.innerWidth();
        const gap = 125;
        const amountValues: number = Math.round(sliderWidth / gap);
        const gapValues: number = sliderWidth / amountValues;
        const valuesArr: number[] = [];
        const pixelsArr: number[] = [];

        for (let i = 0; i < sliderWidth + gapValues; i += gapValues) {
          valuesArr.push(
            convertToNumber(i, sliderWidth, this.options.min, this.options.max)
          );
          pixelsArr.push(i);
        }
        this.rulers = new CreateRulers(
          this.bar.barElement,
          valuesArr,
          pixelsArr
        );
      }
    } else if (this.rulers) {
      this.rulers.rulersElement.remove();
      this.rulers = null;
    }

    if (this.options.configPanel) {
      if (!this.config) {
        this.config = new CreateConfig(
          this.app,
          this.options.toggleConfig,
          this.options.controlConfig
        );
      }

      this.options.controlConfig.forEach((name) => {
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        // @ts-ignore
        const value: number = this.options[name];
        $(`.slider-app__control--${name}`).val(value);
      });

      this.options.toggleConfig.forEach((name) => {
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        // @ts-ignore
        const value: string = this.options[name];
        $(`.slider-app__toggle--${name}`).attr("checked", value);
      });

      if (!this.options.range) {
        $(".slider-app__control--to").prop("disabled", true);
      } else {
        $(".slider-app__control--to").prop("disabled", false);
      }
    } else if (this.config) {
      this.config.configElement.remove();
      this.config = null;
    }
  }
}

export default View;
