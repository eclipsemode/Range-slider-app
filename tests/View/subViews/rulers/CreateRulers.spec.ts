import $ from "jquery";
import { CreateRulers } from "../../../../src/View/subViews";

describe("Create Rulers tests", () => {
  const body: JQuery = $("body");
  const valuesArr: number[] = [1, 10, 20, 30, 40, 50];
  const pixelsArr: number[] = [1, 10, 20, 30, 40, 50];
  let rulers: CreateRulers;

  beforeEach(() => {
    rulers = new CreateRulers(body, valuesArr, pixelsArr, false);
  });
  test("Should be defined", () => {
    expect(rulers.rulersElement).toBeDefined();
  });
  test("Should not match snapshot vertical", () => {
    expect(rulers.rulersElement[0].innerHTML).toMatchSnapshot(
      "slider-app__rulers"
    );
  });
  test("Should not to be null", () => {
    expect(rulers.rulersElement).not.toBeNull();
  });
  test("Should match values", () => {
    expect(rulers.rulersElement[0].innerHTML).toMatch("20");
  });
});
