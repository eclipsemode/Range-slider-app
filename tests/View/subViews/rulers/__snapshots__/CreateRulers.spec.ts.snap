// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Rulers tests Should not match snapshot vertical: slider-app__rulers 1`] = `"<div class=\\"slider-app__rulers-value slider-app__rulers-value--1\\" style=\\"left: 1px;\\">1</div><div class=\\"slider-app__rulers-value slider-app__rulers-value--10\\" style=\\"left: 10px;\\">10</div><div class=\\"slider-app__rulers-value slider-app__rulers-value--20\\" style=\\"left: 20px;\\">20</div><div class=\\"slider-app__rulers-value slider-app__rulers-value--30\\" style=\\"left: 30px;\\">30</div><div class=\\"slider-app__rulers-value slider-app__rulers-value--40\\" style=\\"left: 40px;\\">40</div><div class=\\"slider-app__rulers-value slider-app__rulers-value--50\\" style=\\"left: 50px;\\">50</div>"`;
